cfbs/masterfiles/generate_vcf_git_checkout.py

```diff
@@ -0,0 +1,106 @@
+import os
+import shutil
+import subprocess
+import sys
+
+from cfbs.utils import write_json
+from cfbs.masterfiles.analyze import (
+    finalize_vcf,
+    initialize_vcf,
+    versions_checksums_files,
+)
+
+DIR_PATH = "."
+"""The path of the working directory."""
+
+MPF_URL = "https://github.com/cfengine/masterfiles"
+MPF_PATH = os.path.join(DIR_PATH, "masterfiles")
+
+
+def check_required_command(command):
+    if not shutil.which(command):
+        print("`%s` was not found" % command)
+        sys.exit(1)
+
+
+def check_required_commands(commands):
+    for c in commands:
+        check_required_command(c)
+
+
+def generate_vcf_git_checkout(interesting_tags=None):
+    required_commands = ["git", "make", "automake", "autoconf"]
+    check_required_commands(required_commands)
+
+    # get the current version of the MPF repo
+    if not os.path.isdir(MPF_PATH):
+        subprocess.run(
+            ["git", "clone", "--no-checkout", MPF_URL],
+            cwd=DIR_PATH,
+            check=True,
+        )
+    else:
+        subprocess.run(
+            ["git", "fetch", "--all"],
+            cwd=DIR_PATH,
+            check=True,
+        )
+
+    result = subprocess.run(
+        ["git", "tag"], cwd=MPF_PATH, capture_output=True, check=True
+    )
+    tags = result.stdout.decode("UTF-8").splitlines()
+
+    # if not given, choose tags to checkout - by default, only consider version releases
+    if interesting_tags is None:
+        interesting_tags = []
+
+        for tag in tags:
+            if "-" not in tag:
+                interesting_tags.append(tag)
+
+    versions_dict, checksums_dict, files_dict = initialize_vcf()
+
+    for tag in interesting_tags:
+        print("Checkouting tag", tag)
+
+        # checkout the version
+        subprocess.run(
+            ["git", "checkout", tag],
+            cwd=MPF_PATH,
+            check=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+
+        # build masterfiles from git as they are in the tarball packages
+        subprocess.run(
+            ["./autogen.sh"],
+            cwd=MPF_PATH,
+            check=True,
+            env=dict(os.environ.copy(), EXPLICIT_VERSION=tag),
+        )
+        # older masterfiles version READMEs instruct to use `make install` and newer `make` - always use `make` instead
+        subprocess.run(["make"], cwd=MPF_PATH, check=True)
+
+        # compute VCF data for all the files
+        versions_dict, checksums_dict, files_dict = versions_checksums_files(
+            MPF_PATH, tag, versions_dict, checksums_dict, files_dict
+        )
+
+        # clean the files to prevent spillage to other versions
+        subprocess.run(
+            ["git", "clean", "-dffx"],
```

Comment:
I don't think the second `f` is necessary here, since we don't do `git clone --recursive <REPO>`. However, it should not hurt either. So feel free to dismiss.
```suggestion
            ["git", "clean", "-dfx"],
```